src/main.tf

```
@@ -16,6 +16,7 @@ resource "azurerm_redis_cache" "main" {
   minimum_tls_version           = "1.2"
   subnet_id                     = var.azure_virtual_network.data.infrastructure.default_subnet_id
   public_network_access_enabled = false
+  enable_non_ssl_port           = var.redis.non_ssl_port
```

```suggestion
  enable_non_ssl_port           = try(var.redis.non_ssl_port, false)
```
Since this is a new field, if people decommission before saving/deploying it will fail. Anytime we add a field we need to use a `lookup` or `try`.